test/Datadog.Trace.ClrProfiler.IntegrationTests/MsmqTests.cs

```diff
@@ -0,0 +1,108 @@
+#if NETFRAMEWORK
+using System;
+using System.Linq;
+using Datadog.Core.Tools;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class MsmqTests : TestHelper
+    {
+        private const string ExpectedServiceName = "Samples.Msmq-msmq";
+
+        public MsmqTests(ITestOutputHelper output)
+            : base("Msmq", output)
+        {
+            SetServiceVersion("1.0.0");
+        }
+
+        [Trait("Category", "EndToEnd")]
+        [Fact]
+        public void SubmitTraces()
+        {
+            SetCallTargetSettings(true);
+
+            const int expectedTransactionalTraces = 12;
+            const int expectedNonTransactionalTracesTraces = 12;
+            const int totalTransactions = expectedTransactionalTraces + expectedNonTransactionalTracesTraces;
+            const int expectedPurgeCount = 2;
+            const int expectedPeekCount = 2;
+            const int expectedSendCount = 10;
+            const int expectedReceiveCount = 10;
+
+            var sendCount = 0;
+            var peekCount = 0;
+            var receiveCount = 0;
+            var purgeCount = 0;
+            var transactionalTraces = 0;
+            var nonTransactionalTraces = 0;
+
+            var agentPort = TcpPortProvider.GetOpenPort();
+            using var agent = new MockTracerAgent(agentPort);
+            using var processResult = RunSampleAndWaitForExit(agent.Port, arguments: $"5 5");
+            Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+
+            var spans = agent.WaitForSpans(totalTransactions);
+            Assert.True(spans.Count >= totalTransactions, $"Expecting at least {totalTransactions} spans, only received {spans.Count}");
+            var msmqSpans = spans.Where(span => string.Equals(span.Service, ExpectedServiceName, StringComparison.OrdinalIgnoreCase));
+            foreach (var span in msmqSpans)
+            {
+                Assert.Equal(SpanTypes.Queue, span.Type);
```

Comment:
Just a minor one, but we've shifted towards using FluentAssertions to make this easier to read. e.g. 
```suggestion
                span.Type.Should().Be(SpanTypes.Queue);
```